docs/code-examples/annotation_context_rects.rs

```diff
@@ -21,27 +20,16 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
     )?;
 
     // Log a batch of 2 rectangles with different class IDs
-    // TODO(#2786): Rect2D archetype
-    rec.log_component_batches(
+    rec.log(
         "detections",
-        false,
-        2,
-        [
-            &[
-                Rect2D::XYWH(Vec4D([-2., -2., 3., 3.]).into()),
-                Rect2D::XYWH(Vec4D([0., 0., 2., 2.]).into()),
-            ] as _,
-            &[ClassId::from(1), ClassId::from(2)] as _,
-        ],
+        &Boxes2D::from_xywh([(-2., -2.), (0., 0.)], [(3.0, 3.0), (2.0, 2.0)])
+            .with_class_ids([1, 2]),
     )?;
 
     // Log an extra rect to set the view bounds
-    // TODO(#2786): Rect2D archetype
-    rec.log_component_batches(
+    rec.log(
         "bounds",
-        false,
-        1,
-        [&[Rect2D::XCYCWH(Vec4D([0.0, 0.0, 5.0, 5.0]).into())] as _],
+        &Boxes2D::new([(2.5, 2.5)]).with_centers([(2.5, 2.5)]),
```

Comment:
Same here. This is not very clear code. `Boxes2D::from_center_and_half_extents(…)` is a bit verbose maybe, but at least clear.